### PR TITLE
make removeOptionalTags more conservative

### PR DESCRIPTION
**Pull request by alexlamsl**
As discovered in #503, we should treat all HTML tags around non-standard tags as mandatory.

They might get transformed by JavaScript libraries into various locations of the final DOM tree, which may or may not break the optional tag rules.

### CODE REVIEW

**Comment by alexlamsl** [comment]
rebased

**Comment by alexlamsl** [comment]
rebased again :ghost:

**Comment by alexlamsl** [comment]
@kangax if you would publish `1.2.1` after this, I'd be grateful (to get beaten up by a new round of bug reports... :sweat_smile:)

File: tests/minifier.js [hunk 1]
```diff
@@ -840,6 +840,54 @@
...
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body></body> </html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body></body></html> ';
+    output = ' ';
```
Comment (kangax) [inline tests/minifier.js:886]: Hm, it's probably more theoretical than anything, but I wonder if we really care about leaving space in cases like this...

Comment (alexlamsl) [inline tests/minifier.js:886]: These tests are for the benefit of keeping track of this tricky logic than of any real world importance, I agree.

Pretty sure most people can't tell an empty screen from one with a whitespace on it :ghost:

**Pull request merged**